zerver/lib/export.py

```diff
@@ -2038,6 +2038,43 @@ def get_single_user_config() -> Config:
     return user_profile_config
 
 
+def get_id_list_gently_from_database(*, base_query: Any, id_field: str) -> List[int]:
+    """
+    Use this function if you need a HUGE number of ids from
+    the database, and you don't mind a few extra trips.  Particularly
+    for exports, we don't really care about a little extra time
+    to finish the export--the much bigger concern is that we don't
+    want to overload our database all at once, nor do we want to
+    keep a whole bunch of Django objects around in memory.
+
+    So our general process is to call this function first, and then
+    we call chunkify to break our ids into small chunks for "fat query"
+    batches.
+
+    Even if you are not working at huge scale, this function can
+    also be used for the convenience of its API.
+    """
+    min_id = -1
+    all_ids = []
+    batch_size = 50000  # we are just getting ints
```

Comment:
I'd do 10K; that's been very successful with our database migrations to avoid potentially holding locks for a significant time window.

Reply:
done